src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -0,0 +1,100 @@
+package gov.cabinetoffice.gap.adminbackend.services;
+
+import gov.cabinetoffice.gap.adminbackend.config.UserServiceConfig;
+import gov.cabinetoffice.gap.adminbackend.dtos.schemes.SchemeEditorsDTO;
+import gov.cabinetoffice.gap.adminbackend.dtos.user.UserEmailRequestDto;
+import gov.cabinetoffice.gap.adminbackend.dtos.user.UserEmailResponseDto;
+import gov.cabinetoffice.gap.adminbackend.entities.GrantAdmin;
+import gov.cabinetoffice.gap.adminbackend.entities.SchemeEntity;
+import gov.cabinetoffice.gap.adminbackend.enums.SchemeEditorRoleEnum;
+import gov.cabinetoffice.gap.adminbackend.repositories.SchemeRepository;
+import gov.cabinetoffice.gap.adminbackend.services.encryption.AwsEncryptionServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class SchemeEditorService {
+    private final RestTemplate restTemplate;
+    private final SchemeService schemeService;
+    private final AwsEncryptionServiceImpl awsEncryptionService;
+    private final SchemeRepository schemeRepo;
+    private final UserServiceConfig userServiceConfig;
+
+    @Value("${user-service.domain}")
+    private String userServiceUrl;
+
+    public Boolean doesAdminOwnScheme(Integer schemeId, Integer adminId) {
+        SchemeEntity scheme = schemeService.findSchemeById(schemeId);
+        List<SchemeEntity> adminSchemes = schemeRepo.findByCreatedBy(adminId);
+        if (adminSchemes.contains(scheme)) {
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
```

Comment:
I think we can refactor this whole function to be a lot more efficient with a single DB call:
```
    boolean existsByIdAndGrantAdminsId(Integer schemeId, Integer grantAdminId);
```

Reply:
very nice